alica_engine/src/engine/UtilityFunction.cpp

```diff
@@ -224,16 +218,13 @@ void UtilityFunction::init(AlicaEngine* ae)
             } else {
                 curPrio = iter->second;
             }
-            TaskRoleStruct* trs = new TaskRoleStruct(taskId, roleId);
-            if (this->priorityMartix.find(trs) == this->priorityMartix.end()) {
-                this->priorityMartix.insert(pair<TaskRoleStruct*, double>(trs, curPrio));
-            }
+            priorityMatrix[TaskRoleStruct(taskId, roleId)] = curPrio;
```

Comment:
It looks like the behavior change from ignoring duplicate entries to overwriting them. Is that intentional / what are the consequences of this?

Reply:
yes that is intentional. Basically, that would indicate bad data. The last value is as good as any.